torch/distributions/transforms.py

```diff
@@ -185,6 +179,34 @@ def log_abs_det_jacobian(self, x, y):
     def __repr__(self):
         return self.__class__.__name__ + '()'
 
+    def forward_shapes(self, batch_shape, event_shape):
+        """
+        Infers the batch and event shapes of the forward computation.
+        Defaults to preserving total shape but ensuring minimum event_dim.
```

Comment:
I am not sure why we need to forward both `batch_shape` and `event_shape`? Shouldn't `forward_shapes(shape)` enough?

Reply:
Well we need helpers to compute one of the following representations of metadata:
either `(batch_shape, event_shape)` or `(shape, event_dim)`. I have opted for `(batch_shape, event_shape)` since that fits in a single function. That representation also makes the `TransformedDistribution` computations nice. I suppose we could change `.forward_shape()` to compute directly on `shape`, but then in most places we use it we would need to manually examine `.domain.event_dim` or `.codomain.event_dim`. 🤷

Reply:
Let me code up your suggested `.forward_shape(shape)` version to see if it's any simpler...

Reply:
@fehiepsi you're right, `.forward_shape(shape)` is much simpler. I just pushed. Thanks 😄